record/program/src/instruction.rs

```diff
@@ -52,19 +53,37 @@ pub enum RecordInstruction<'a> {
     /// 1. `[signer]` Record authority
     /// 2. `[]` Receiver of account lamports
     CloseAccount,
+
+    /// Reallocate additional space in a record account
+    ///
+    /// If the record account already has enough space to hold the specified
+    /// data length, then the instruction does nothing.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    /// 0. `[writable]` The record account to reallocate
+    /// 1. `[signer, writable]` The payer account to fund reallocation
+    /// 2. `[]` System program for reallocation funding
```

Comment:
Rather than passing these accounts in, how about we have the client library top up the account with the new required rent exemption?

Reply:
Okay, yes agreed. I removed the lamports logic! Let me know what you think.